src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/offhand/ModuleOffhand.kt

```diff
@@ -0,0 +1,373 @@
+            inner.remove(switchMode)
+        }
+    }
+
+    private val INVENTORY_MAIN_PRIORITY = INVENTORY_SLOTS + HOTBAR_SLOTS
+    private val INVENTORY_HOTBAR_PRIORITY = HOTBAR_SLOTS + INVENTORY_SLOTS
+    private val chronometer = Chronometer()
+    private var activeMode: Mode = Mode.NONE
+    private var lastMode: Mode? = null
+    private var lastTagMode: Mode = Mode.NONE
+    private var staticMode = Mode.NONE
+    private var last: Pair<Item, ItemSlot>? = null
+
+    override val tag: String
+        get() = activeMode.modeName
+
+    override fun enable() {
+        staticMode = when {
+            Crystal.enabled && Mode.CRYSTAL.canCycleTo() -> Mode.CRYSTAL
+            Gapple.enabled -> Mode.GAPPLE
+            totem.enabled && !Totem.Health.enabled -> Mode.TOTEM
+            else -> Mode.NONE
+        }
+    }
+
+    @Suppress("unused")
+    val keyHandler = handler<KeyEvent> {
+        if (it.action != GLFW.GLFW_PRESS) {
+            return@handler
+        }
+
+        when (it.key.keyCode) {
+            Gapple.gappleBind -> Mode.GAPPLE.onBindPress()
+            Crystal.crystalBind -> Mode.CRYSTAL.onBindPress()
+            Strength.strengthBind -> {
+                // since we can't cycle to strength, its status has to be checked here
+                if (Strength.enabled) {
+                    Mode.STRENGTH.onBindPress()
+                }
+            }
+
+            cycleSlots -> {
+                val entries = Mode.entries
+                val startIndex = staticMode.ordinal
+                var index = (startIndex + 1) % entries.size
+
+                while (index != startIndex) {
+                    val mode = entries[index]
+                    if (mode.canCycleTo()) {
+                        staticMode = mode
+                        return@handler
+                    }
+
+                    index = (index + 1) % entries.size
+                }
+            }
+        }
+    }
+
+    @Suppress("unused")
+    private val autoTotemHandler = handler<ScheduleInventoryActionEvent>(priority = 100) {
+        activeMode = Mode.entries.firstOrNull(Mode::shouldEquip) ?: staticMode
+        if (activeMode == Mode.NONE && Totem.Health.switchBack && lastMode == Mode.TOTEM) {
+            activeMode = Mode.BACK
+        }
+
+        if (activeMode != lastTagMode) {
+            EventManager.callEvent(RefreshArrayListEvent())
+            lastTagMode = activeMode
+        }
+
+        if (!chronometer.hasElapsed(activeMode.getDelay().toLong())) {
+            return@handler
+        }
+
+        val slot = activeMode.getSlot() ?: return@handler
+        lastMode = activeMode
+
+        // the item is already located in Off-hand slot
+        if (slot == OFFHAND_SLOT) {
+            return@handler
+        }
+
+        if (Totem.Health.switchBack) {
+            last = slot.itemStack.item to slot
+        }
+
+        val actions = if (hasProtocolTranslator) {
+            if (isOlderThanOrEquals1_12_2) {
```

Comment:
1.16 added the ability to use the offhand key in the inventory menu

Reply:
I tested that and it did not work reliable.